src/tools/rustfmt/src/parse/session.rs

```diff
@@ -40,6 +41,23 @@ impl Translate for SilentEmitter {
     fn fallback_fluent_bundle(&self) -> &rustc_errors::FluentBundle {
         panic!("silent emitter attempted to translate a diagnostic");
     }
+
+    // Override `translate_message` for the silent emitter because eager translation of
+    // subdiagnostics result in a call to this.
+    fn translate_message<'a>(
```

Comment:
The duplication of this method is unfortunate. Can it be avoided, e.g. by defining it in `rustc_errors` and exporting it?

Reply:
Fixed!